src/handlers/allowedProps.ts

```diff
@@ -0,0 +1,249 @@
+import { PanGestureHandlerProps } from './gestureHandlers';
+
+const commonProps = [
+  'id',
+  'enabled',
+  'minPointers',
+  'shouldCancelWhenOutside',
+  'hitSlop',
+] as const;
+
+const componentInteractionProps = ['waitFor', 'simultaneousHandlers'] as const;
+
+export const baseGestureHandlerProps = [
+  ...commonProps,
+  ...componentInteractionProps,
+  'onBegan',
+  'onFailed',
+  'onCancelled',
+  'onActivated',
+  'onEnded',
+  'onGestureEvent',
+  'onHandlerStateChange',
+] as const;
+
+export const tapGestureHandlerProps = [
+  'maxDurationMs',
+  'maxDelayMs',
+  'numberOfTaps',
+  'maxDeltaX',
+  'maxDeltaY',
+  'maxDist',
+  'minPointers',
+] as const;
+
+export const flingGestureHandlerProps = [
+  'numberOfPointers',
+  'direction',
+] as const;
+
+export const forceTouchGestureHandlerProps = [
+  'minForce',
+  'maxForce',
+  'feedbackOnActivation',
+] as const;
+
+export const longPressGestureHandlerProps = [
+  'minDurationMs',
+  'maxDist',
+] as const;
+
+export const panGestureHandlerProps = [
+  'activeOffsetY',
+  'activeOffsetX',
+  'failOffsetY',
+  'failOffsetX',
+  'minDist',
+  'minVelocity',
+  'minVelocityX',
+  'minVelocityY',
+  'minPointers',
+  'maxPointers',
+  'avgTouches',
+  'enableTrackpadTwoFingerGesture',
+] as const;
+
+export const panGestureHandlerCustomNativeProps = [
+  'activeOffsetYStart',
+  'activeOffsetYEnd',
+  'activeOffsetXStart',
+  'activeOffsetXEnd',
+  'failOffsetYStart',
+  'failOffsetYEnd',
+  'failOffsetXStart',
+  'failOffsetXEnd',
+] as const;
+
+export const nativeViewGestureHandlerProps = [
+  'shouldActivateOnStart',
+  'disallowInterruption',
+] as const;
+
+function validatePanGestureHandlerProps(props: PanGestureHandlerProps) {
```

Comment:
I think this function should be in `createHandler.ts` file since it's used only there and its sole purpose is to validate Pan props.

Reply:
Since `gestureHandlers.ts` is being split into separate files for each handler, can we put those methods in the same file as `PanGestureHandler`? If I have the same idea about splitting, then that will be the only place it will be used.

Reply:
Yup, I wrote this comment before comment on review.

Reply:
Done in e80d121cf21ee01911a14ec4ac83e6dd889ef436